framework/py/flwr/superexec/exec_license_interceptor.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2025 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower Exec API license interceptor."""
+
+
+from collections.abc import Iterator
+from typing import Any, Callable, Union
+
+import grpc
+from google.protobuf.message import Message as GrpcMessage
+
+from flwr.supercore.license_plugin import LicensePlugin
+
+
+class ExecLicenseInterceptor(grpc.ServerInterceptor):  # type: ignore
+    """Exec API interceptor for license checking."""
+
+    def __init__(self, license_checker: LicensePlugin) -> None:
```

Comment:
Let's name this consistently everywhere:
```suggestion
    def __init__(self, license_plugin: LicensePlugin) -> None:
```